spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Perf.scala

```diff
@@ -86,34 +86,39 @@ object Perf {
 
       val criterion = CrossEntropyCriterion()
 
-      if (training) {
-        if (model.isInstanceOf[MklDnnContainer]) {
-          model.asInstanceOf[MklDnnContainer]
-            .compile(TrainingPhase, Array(HeapData(inputShape, inputFormat)))
-        } else if (model.isInstanceOf[DnnGraph]) {
-          model.asInstanceOf[DnnGraph].compile(TrainingPhase)
-        }
-        model.training()
-      } else {
-        if (model.isInstanceOf[MklDnnContainer]) {
-          model.asInstanceOf[MklDnnContainer]
-            .compile(InferencePhase, Array(HeapData(inputShape, inputFormat)))
-        } else if (model.isInstanceOf[DnnGraph]) {
-          model.asInstanceOf[DnnGraph].compile(InferencePhase)
+      Engine.computing.invokeAndWait2(Array(1).map(_ => () => {
+        if (training) {
+          if (model.isInstanceOf[MklDnnContainer]) {
```

Comment:
what's difference between dnngraph and mkldnngraph ?

Reply:
DnnGraph inherits from StaticGraph directly. And is not the type of MklDnnContainer.